kong/plugins/key-auth/handler.lua

```diff
@@ -149,6 +150,15 @@ end
 function KeyAuthHandler:access(conf)
   KeyAuthHandler.super.access(self)
 
+  -- check if preflight request and wether it should be authenticated
+  if conf.authenticate_preflight == false and get_method() == "OPTIONS" then
```

Comment:
Instead of making a "default true" option (`authenticate_preflight`) we could make a "default false" one (`skip_preflight`). That way the default nil would just work: `if conf.skip_preflight and ...`. No migration needed.